### PR TITLE
Refactor ImportDatabaseConfigurationHandler.

### LINKED ISSUE #18746: Refactor `ImportDatabaseConfigurationHandler` to only allow imports when database is empty
Hi community,

We have a pair of statements in DistSQL for export and import configuration, they are:
- EXPORT DATABASE CONFIGURATION
- IMPORT DATABASE CONFIGURATION
> https://github.com/apache/shardingsphere/pull/15069
https://github.com/apache/shardingsphere/pull/16162

Among them, there may be some problems in the processing of `IMPORT`:
1. When the data sources already exist in the metadata, the current practice is to merge the imported ones, but the removed resources are not well closed, which may lead to connection leakage;
2. When a configuration already exists, it is difficult to tell whether the user wants to merge or replace completely;
3. Refer to some other databases, hope the target is clean when importing, such API is easier to understand.

Therefore, I recommend adjusting the implementation of `ImportDatabaseConfigurationHandler` to only allow imports if the current database's configuration is empty.

If anyone wants to try it, please leave a comment on this issue.

### PR DESCRIPTION
**Pull request by yx9o**
Fixes #18746 .

Changes proposed in this pull request:
- Refactor ImportDatabaseConfigurationHandler to only allow imports when database is empty.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java [hunk 1]
```diff
@@ -94,73 +91,84 @@ protected void update(final ContextManager contextManager) throws DistSQLExcepti
...
         checkDataSource(yamlConfig.getDataSources(), file);
-        updateResources(databaseName, yamlConfig.getDataSources());
-        updateRules(databaseName, yamlConfig.getRules());
+        addDatabase(databaseName);
+        addResources(databaseName, yamlConfig.getDataSources());
+        try {
+            addRules(databaseName, yamlConfig.getRules());
+        } catch (DistSQLException ex) {
+            dropDatabase(databaseName);
+            throw new RuntimeException(ex.getMessage());
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java:100]: Why is RuntimeException thrown?

Comment (yx9o) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java]: I made adjustments, please review, thank you.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java [hunk 2]
```diff
@@ -94,73 +93,84 @@ protected void update(final ContextManager contextManager) throws DistSQLExcepti
...
+            throw ex;
+        }
     }
     
     private void checkDatabaseName(final String databaseName, final File file) {
         Preconditions.checkNotNull(databaseName, String.format("Property `databaseName` in file `%s` is required.", file.getName()));
-        if (!ProxyContext.getInstance().getAllDatabaseNames().contains(databaseName)) {
-            throw new DatabaseNotExistedException(databaseName);
+        if (ProxyContext.getInstance().getAllDatabaseNames().contains(databaseName)) {
+            Preconditions.checkState(ProxyContext.getInstance().getDatabase(databaseName).getResource().getDataSources().isEmpty(), "Database `%s` exists.", databaseName);
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java:109]: The dataSources are checked here, but the message is 'Database `%s` exists', which is not easy to understand.

Comment (yx9o) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java]: Hi, I made adjustments, please review, thank you.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**